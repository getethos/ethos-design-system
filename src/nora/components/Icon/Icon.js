import PropTypes from 'prop-types'
import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export const Icon = ({ iconPrefix, iconName, className }) => {
  return (
    <FontAwesomeIcon className={className} icon={[iconPrefix, iconName]} />
  )
}

Icon.propTypes = {
  iconPrefix: PropTypes.string.isRequired,
  iconName: PropTypes.string.isRequired,
  className: PropTypes.string,
}
