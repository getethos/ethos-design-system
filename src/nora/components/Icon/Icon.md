Icon is a wrapper around the FontAwesomeIcon component.
If you would like to add an icon that does not currently exist, please update fa.js.

---

Default Icon component

```jsx
<Icon iconPrefix="fal" iconName="file-medical-alt" />
```

Icon component with custom styling

```jsx
import styles from './IconExample.module.scss';
<Icon
  iconPrefix="fal"
  iconName="file-medical-alt"
  className={styles.IconClassName}
/>
```
