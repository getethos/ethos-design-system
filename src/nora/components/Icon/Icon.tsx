import { IconName, IconPrefix } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as React from 'react'

type IconProps = {
  iconPrefix: string
  iconName: string
  className?: string
}
export const Icon: React.SFC<IconProps> = ({
  iconPrefix,
  iconName,
  className,
}) => {
  return (
    <FontAwesomeIcon
      className={className}
      icon={[iconPrefix as IconPrefix, iconName as IconName]}
    />
  )
}
