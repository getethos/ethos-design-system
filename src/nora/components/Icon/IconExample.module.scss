@import '../Colors.css';

.IconClassName {
  color: var(--NoraBlue-400);
}