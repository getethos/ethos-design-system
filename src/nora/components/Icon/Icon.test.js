import React from 'react'
import renderer from 'react-test-renderer'
import { Icon } from './Icon'

describe('Icon', () => {
  it('renders default Icon', () => {
    const tree = renderer
      .create(<Icon iconPrefix="fal" iconName="file-medical-alt" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
