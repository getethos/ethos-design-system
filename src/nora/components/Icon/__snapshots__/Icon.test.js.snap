// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders default Icon 1`] = `null`;
